### PR TITLE
Enable redirection and fix for subfolders

### LINKED ISSUE #2: Redirecting subfolders
Currently visiting `/docs` shows the content of `/docs/installation.md`.
However visiting `/docs/subdoc` does not show the content of `/docs/subdoc/installation.md`.

Would that be ideal?

Also, would it be possible to direct from `/docs` to `/docs/installation.md` instead of showing its content?

Structure:
```
.
+-- docs
|   +-- installation.md
|   +-- documentation.md
|   +-- subdoc
|   |   +-- installation.md
|   |   +-- documentation.md
```

If this is approved I can try make a pull request for this with configurations.

### PR DESCRIPTION
**Pull request by marktopper**
Fixes #2 

- [x] Allow redirection by configuration
- [x] Make subfolders work
- [x] Add support for folders without index
- [x] Add support for fallback indexes (`docs.index = getting-started,installation` will first try `getting-started` as index, and if that does not exists, it will try to use `installation` as index.

### CODE REVIEW

**Comment by marktopper** [comment]
This pull request is ready for your review.

**Comment by marktopper** [comment]
Let me know if you have any questions @znck

**Comment by marktopper** [comment]
Looking forward for your review @znck 👍

File: src/Docs/DocsBuilder.php [hunk 1]
```diff
@@ -117,7 +117,7 @@ public function build(array $files, array $data)
                 }] + $options;
 
             if (config('docs.redirect')) {
-                $filename = head(explode('.', $target));
+                $filename = substr($target, 0, -strlen('.md'));
```
Comment (znck) [inline src/Docs/DocsBuilder.php]: 👍

File: src/Docs/DocsBuilder.php [hunk 2]
```diff
@@ -138,9 +138,12 @@ protected function getLandings(array $docs)
...
 
         foreach ($docs as $doc) {
-            $paths[] = $doc->getRelativePath();
+            $relativePath = $doc->getRelativePath();
+            if (!in_array($relativePath, $paths)) {
+                $paths[] = $doc->getRelativePath();
+            }
 
-            $name = head(explode('.', $doc->getBasename()));
+            $name = head(explode('.', $doc->getBasename(), 2));
```
Comment (znck) [inline src/Docs/DocsBuilder.php]: If feel you should use substring here too (same as L120).

**Review by znck — COMMENTED** [review]
LGTM but I want to test it before merging. ~

> 😑  Now its perfect time to write covering tests (I am on it). Spending this weekend on Sereno. 💯

File: src/Docs/DocsBuilder.php (hunk 2 above)
Comment (marktopper) [reply src/Docs/DocsBuilder.php]: Thanks, done. 👍

**Comment by marktopper** [comment]
Sorry to hear that you have to use your weekend on that 😑

**Comment by znck** [comment]
Thanks a lot 👍

**Comment by marktopper** [comment]
@znck Could you do a patch release?

**Pull request merged**